diff --git a/internal/core/src/index/InvertedIndexTantivy.cpp b/internal/core/src/index/InvertedIndexTantivy.cpp
index 7b45f32..7be6d3b 100644
--- a/internal/core/src/index/InvertedIndexTantivy.cpp
+++ b/internal/core/src/index/InvertedIndexTantivy.cpp
@@ -433,7 +433,7 @@ InvertedIndexTantivy<T>::BuildWithRawDataForUT(size_t n,
             // only used in ut.
             auto arr = static_cast<const boost::container::vector<T>*>(values);
             for (size_t i = 0; i < n; i++) {
-                wrapper_->template add_multi_data(
+                wrapper_->add_multi_data(
                     arr[i].data(), arr[i].size(), i);
             }
         } else {
@@ -444,7 +444,7 @@ InvertedIndexTantivy<T>::BuildWithRawDataForUT(size_t n,
             // only used in ut.
             auto arr = static_cast<const boost::container::vector<T>*>(values);
             for (size_t i = 0; i < n; i++) {
-                wrapper_->template add_multi_data_by_single_segment_writer(
+                wrapper_->add_multi_data_by_single_segment_writer(
                     arr[i].data(), arr[i].size());
             }
         } else {
@@ -551,12 +551,12 @@ InvertedIndexTantivy<T>::build_index_for_array(
             }
             auto length = data->is_valid(i) ? array_column[i].length() : 0;
             if (!inverted_index_single_segment_) {
-                wrapper_->template add_multi_data(
+                wrapper_->add_multi_data(
                     reinterpret_cast<const T*>(array_column[i].data()),
                     length,
                     offset++);
             } else {
-                wrapper_->template add_multi_data_by_single_segment_writer(
+                wrapper_->add_multi_data_by_single_segment_writer(
                     reinterpret_cast<const T*>(array_column[i].data()), length);
             }
         }
@@ -585,10 +585,10 @@ InvertedIndexTantivy<std::string>::build_index_for_array(
             }
             auto length = data->is_valid(i) ? output.size() : 0;
             if (!inverted_index_single_segment_) {
-                wrapper_->template add_multi_data(
+                wrapper_->add_multi_data(
                     output.data(), length, offset++);
             } else {
-                wrapper_->template add_multi_data_by_single_segment_writer(
+                wrapper_->add_multi_data_by_single_segment_writer(
                     output.data(), length);
             }
         }
